### PR TITLE
Improve Windows ARM64 building & packaging

### PR DESCRIPTION
**Pull request by skyline75489**
A major overhaul of the NuGet packaging pipeline, including:

* Restructure capi & NuGet packaging pipeline to make multi-platform nupkg possible. The NuGet artifacts are separated by EP(CPU, Cuda and DirectML). A single nupkg contains native libraries for different os & arch.
* Make win-arm64 CPU packages use DLLs from NuGet package, just like win-x64.
* Enable win-arm64 DirectML NuGet build & packaging (Internal work item:  https://task.ms/aii/40037)
* Enable linux-x64 CPU & CUDA NuGet build & packaging (Internal work item:  https://task.ms/aii/34234)
* Continue the work of #683 to make it work with ARM64.

Run: https://dev.azure.com/aiinfra/ONNX%20Runtime/_build/results?buildId=507547&view=results

Prior work: #498 
Supports: #637

 Future work: Needs #693 to fix Linux NuGet.

### CODE REVIEW

File: tools/nuget/generate_nuspec_for_native_nuget.py [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+    runtimes = ["win-x64", "win-arm64", "linux-x64"]
+    for runtime in runtimes:
+      if runtime.startswith("win"):
+          add_native_artifact_if_exists(lines, runtime, "onnxruntime-genai.lib")
+          add_native_artifact_if_exists(lines, runtime, "onnxruntime-genai.dll")
+      if runtime.startswith("linux"):
+          add_native_artifact_if_exists(lines, runtime, "libonnxruntime-genai.so")
+
+    # targets
+    for dotnet in ["netstandard2.0", "net8.0", "native"]:
```
Comment (baijumeswani) [inline tools/nuget/generate_nuspec_for_native_nuget.py:132]: Do we need to put the targets file in native also? I don't think we need to do that.

Comment (skyline75489) [reply tools/nuget/generate_nuspec_for_native_nuget.py:132]: This is actually from the original nuspec:

https://github.com/microsoft/onnxruntime-genai/blob/cb313b49adec51f8679701df7d2451c9d21374a2/nuget/Microsoft.ML.OnnxRuntimeGenAI.nuspec#L53

I'm no NuGet expert so I try to keep the original nuspec intact.

**Review by baijumeswani — APPROVED** [review]
Looks good. Thanks for making this work.

File: .pipelines/stages/jobs/capi-packaging-job.yml [hunk 2]
```diff
@@ -0,0 +1,136 @@
...
+      value: 'Microsoft.ML.OnnxRuntime.DirectML'
+    ${{ elseif eq(parameters.ep, 'rocm')}}:
+      value: 'Microsoft.ML.OnnxRuntime.ROCm'
+    ${{ else }}:
+      value: 'Microsoft.ML.OnnxRuntime'
+
+  - name: ortHome
+    value: 'ort'
+  - name: dml_dir
+    value: 'Microsoft.AI.DirectML.1.14.2'
```
Comment (snnn) [inline .pipelines/stages/jobs/capi-packaging-job.yml:80]: They just upgraded this version to 1.15.0

File: .pipelines/stages/jobs/nuget-packaging-job.yml [hunk 3]
```diff
@@ -44,125 +61,184 @@ jobs:
...
+    displayName: 'Extract Artifacts & Prepare Native Libraries'
+    workingDirectory: '$(Build.BinariesDirectory)/artifact-downloads'
+
+  - task: UsePythonVersion@0
+    inputs:
+      versionSpec: 3.10
+      addToPath: true
+
+  - powershell: |
+      python3.exe $(Build.Repository.LocalPath)\tools\nuget\generate_nuspec_for_native_nuget.py `
```
Comment (snnn) [inline .pipelines/stages/jobs/nuget-packaging-job.yml:205]: python.exe

Comment (skyline75489) [reply .pipelines/stages/jobs/nuget-packaging-job.yml]: fixed

File: .pipelines/stages/jobs/capi-packaging-job.yml (hunk 2 above)
Comment (skyline75489) [reply .pipelines/stages/jobs/capi-packaging-job.yml]: done

**Pull request merged**